resource_booking/models/resource_booking_combination.py

```diff
@@ -0,0 +1,101 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.addons.resource.models.resource import Intervals
+
+
+class ResourceBookingCombination(models.Model):
+    _name = "resource.booking.combination"
+    _description = "Bookable resource combinations"
+
+    active = fields.Boolean(index=True, default=True)
+    booking_count = fields.Integer(
+        compute="_compute_booking_count", string="Booking count"
+    )
+    booking_ids = fields.One2many(
+        comodel_name="resource.booking",
+        inverse_name="combination_id",
+        string="Bookings",
+    )
+    forced_calendar_id = fields.Many2one(
+        comodel_name="resource.calendar",
+        string="Forced calendar",
+        index=True,
+        help="Force a specific calendar, instead of combining the resources'.",
+    )
+    name = fields.Char(compute="_compute_name", store=True)
+    type_count = fields.Integer(compute="_compute_type_count", string="Booking types")
+    type_rel_ids = fields.One2many(
+        comodel_name="resource.booking.type.combination.rel",
+        inverse_name="combination_id",
+        string="Resource booking types",
+        help="Resource booking types where this combination is available.",
+    )
+    resource_ids = fields.Many2many(
+        string="Resources",
+        comodel_name="resource.resource",
+        required=True,
+        help="Resources that must be free to be booked together.",
+    )
+
+    @api.depends("booking_ids")
+    def _compute_booking_count(self):
+        for one in self:
+            one.booking_count = len(one.booking_ids)
+
+    @api.depends("resource_ids.name", "forced_calendar_id.name")
+    def _compute_name(self):
+        for one in self:
+            data = {
+                "resources": " + ".join(sorted(one.resource_ids.mapped("name"))),
+                "calendar": one.forced_calendar_id.name,
+            }
+            if one.forced_calendar_id:
+                one.name = _("%(resources)s (using calendar %(calendar)s)") % data
+            else:
+                one.name = _("%(resources)s") % data
+
+    @api.depends("type_rel_ids")
+    def _compute_type_count(self):
+        for one in self:
+            one.type_count = len(one.type_rel_ids)
+
+    @api.constrains("booking_ids", "forced_calendar_id", "resource_ids")
+    def _check_bookings_scheduling(self):
+        """Scheduled bookings must have no conflicts."""
+        bookings = self.mapped("booking_ids")
+        return bookings._check_scheduling()
+
+    def _get_intervals(self, start_dt, end_dt):
+        """Get available intervals for this booking combination."""
+        base = Intervals([(start_dt, end_dt, self)])
+        result = Intervals([])
+        for combination in self:
+            combination_intervals = base
+            for res in combination.resource_ids:
+                if not combination_intervals:
+                    break  # Can't restrict more
```

Comment:
Why putting this check inside the loop? You could avoid entering that `if` in every iteration by:
```suggestion
        for combination in self:
            combination_intervals = base
            if combination_intervals:
                for res in combination.resource_ids:
```

Reply:
Yes, but `combination_intervals` gets more restricted with each loop (see next lines). If, after some iteration, it becomes void, we don't need to continue checking.

Reply:
Ok, thanks, I hadn't seen that `combination_intervals` changed bellow...